pkg/network/cluster_config.go

```diff
@@ -19,20 +20,33 @@ func ValidateClusterConfig(clusterConfig configv1.NetworkSpec) error {
 	// Check all networks for overlaps
 	pool := iputil.IPPool{}
 
-	if len(clusterConfig.ServiceNetwork) == 0 {
```

Comment:
nit: we could keep this check as is to bail out early. and leave the dual-stackness+count checks for after we've parsed the `clusterConfig.ServiceNetwork`. Although for keeping the count+dual-stackness logic localized, this is fine too.